app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -121,6 +121,24 @@ public function clearShoppingCart() {
         $cartHelper->getCart()->truncate()->save();
     }
 
+    /**
+     * If the session quote has been flagged by having a parent quote Id equal to its own
```

Comment:
why do we have to clear the cache? isn't cart content same?

Reply:
@daisy1754 
It is better to think of this here from the perspective of quotes and orders.  Remember, we are caching Bolt order tokens which are tied to a specific immutable quote.  Regardless of the content of the quote, what we have done prior to the point cancelling the pre-auth order is that we have converted the quote into an order.  Up to this point, we have not experimented with trying to 
- reuse a single quote to create multiple orders
- potentially have two orders exist using the same quote
- reverting a previously converted quote to the unconverted state

There are several things to be cautious of in pursuing the approach of reverting a quote from an order and not producing multiple duplicate orders and it is something to be pursued thoroughly in a future PR, (e.g. removing immutable quotes from our system and using only the native Magento quote.)  

For now, we do not try to over-complicate the issue and concern ourselves with the problem of quote reuse for multiple orders.  Instead, we live with the footprint of a new unused quote in the event of the customer choosing to refresh his browser after a payment failure.  If the customer does not refresh his browser, there will be no problems